spec/editors_draft.html

```diff
@@ -3298,6 +3298,244 @@ <h2>Evaluation Semantics</h2>
+        Allowing multiple such composite values, from multiple sources, to be inserted into the same RDF Dataset (e.g., via two SPARQL `LOAD` updates) presents a challenge:
+        the <a>term lists</a> obtained by applying the <a href="#list-datatype-lex-to-value-mapping">cdt:List lexical-to-value mapping</a> will contain the same blank nodes (because, during the mapping process for each cdt:List literal, the <i><a>bnl2bn</a></i> function is applied to the same blank node identifier, `_:eve`).
+        This is problematic, as it introduces a form of unintentional <a href="https://www.w3.org/TR/rdf11-concepts/#dfn-skolem-iri">skolemization</a>, removing the scoping of serialized blank nodes and allowing them to be universally referenced so long as they are wrapped in an enclosing composite type.
+      </p>
+        
+      <p>
+        The following sections describe requirements that conforming SPARQL systems must comply with to support proper handling of blank nodes when importing or exporting RDF data that contains cdt:List or cdt:Map literals.
+      </p>
+    </section>
+
+    <section>
+      <h2>Importing Requirements</h2>
+      <p>
+        Conforming implementations MUST process <a>cdt:List literal</a>s and <a>cdt:Map literal</a>s during loading, replacing, in their lexical form, any substring _:<var>id</var>
+        matching the <a data-cite="TURTLE#grammar-production-BLANK_NODE_LABEL">`BLANK_NODE_LABEL`</a> production of the grammar
+        with a string "_:"+<var>b</var> where <var>b</var> is a blank node identifier, such that
+      </p>
+      
+      <ul>
+        <li><i><a>bnl2bn</a></i>(<var>b</var>) must be a fresh blank node not already used in the SPARQL system or RDF Dataset, and</li>
+        <li>within the scope of the document being processed, all occurrences of <var>id</var> used as a blank node identifier (both inside and outside of composite values) are replaced with the same <var>b</var> value, and</li>
+        <li>within the scope of the document being processed, two distinct <var>id</var> blank node identifiers are replaced with distinct <var>b</var> values, and</li>
+        <li>
+          if the format of the document being processed has a mechanism for explicitly identifying blank nodes outside of cdt:List and cdt:Map literals (e.g., `_:b1` syntax in N-Triples and in Turtle, `rdf:nodeID` in RDF/XML),
+          and the blank node identifier <var>id</var> appears both inside and outside of such literals,
+          and the parser for this document format emits a blank node <var>B</var> for occurrences of <var>id</var> outside of the literals,
+          then the rewritten blank node identifier <var>b</var> MUST be a lexical form that maps to <var>B</var> in the lexical-to-value mapping (that is, <i><a>bnl2bn</a></i>(<var>b</var>) = <var>B</var>).
+        </li>
+      </ul>
```

Comment:
There are still some issues here:
* There is still an inconsistency in terms of how the first and the last bullet point use the _bnl2bn_ function. That is, these bullet points say "_bnl2bn_(_b_)" where you have introduced _b_ as a blank node identifier, but the _bnl2bn_ function is not meant to be applied to blank node identifiers (but to strings that begin with `_:`, followed by a bnode identifier).
* The first and the last bullet point dependent on one another in the sense that the condition of the first bullet would be implicitly satisfied in cases in which the last bullet point is relevant (i.e., in cases in which the format of the document being processed has a mechanism for explicitly identifying blank nodes). Consequently, I think that the first bullet point needs to be mentioned explicitly only for cases in which the format of the document being processed does *not* have a mechanism for explicitly identifying blank nodes. Do you agree?

Note that I just wanted to list the issues here for the record. I will also think about how to change the text to fix/avoid these issues.

Reply:
Here is a proposal to change this whole part:
```suggestion
      <p>
        Conforming implementations MUST process <a>cdt:List literal</a>s and <a>cdt:Map literal</a>s during loading, replacing, in their lexical form, any substring <var>bnl</var>
        matching the <a data-cite="TURTLE#grammar-production-BLANK_NODE_LABEL">`BLANK_NODE_LABEL`</a> production of the grammar
        with a string <var>bnl'</var> such that
      </p>
      
      <ul>
        <li>
          <var>bnl'</var> also matches the <a data-cite="TURTLE#grammar-production-BLANK_NODE_LABEL">`BLANK_NODE_LABEL`</a> production, and
        </li>
        <li>
          if the format of the document being processed has a mechanism for explicitly identifying blank nodes outside of cdt:List and cdt:Map literals (e.g., `_:b1` syntax in N-Triples and in Turtle, `rdf:nodeID` in RDF/XML),
          and <var>id</var> is the blank node identifier represented by <var>bnl</var> (i.e., <var>bnl</var> is of the form `_:`<var>id</var>),
          then <var>bnl'</var> must be picked such that <i><a>bnl2bn</a></i>(<var>bnl'</var>)=<var>B</var> where <var>B</var> is the blank node that the parser for this document emits for occurrences of <var>id</var> outside of the literals, and
        </li>
        <li>
          if the format of the document being processed does not have a mechanism for explicitly identifying blank nodes outside of cdt:List and cdt:Map literals, then <var>bnl'</var> must be picked such that <i><a>bnl2bn</a></i>(<var>bnl'</var>) is a fresh blank node not already used in the SPARQL system or RDF Dataset, and
        </li>
        <li>within the scope of the document being processed, all occurrences of <var>bnl</var> used to represent a blank node identifier within the lexical forms of cdt:List and cdt:Map literals are replaced with the same <var>bnl'</var> value, and</li>
        <li>within the scope of the document being processed, no other substring that matches the <a data-cite="TURTLE#grammar-production-BLANK_NODE_LABEL">`BLANK_NODE_LABEL`</a> production within the lexical form of a cdt:List or cdt:Map literal is replaced by the same substring <var>bnl'</var>.</li>
        </li>
      </ul>
```